### ISSUE TITLE
Finalize DID document creation form

### ISSUE THREAD

**Issue by xamcost** [Sedimark/marketplace-frontend#4]
Hi there! The goal of this issue is to decide the content of the form presented to prospective users to enroll in Sedimark, prior to the creation of their corresponding DID document and participant Verifiable Credential. We also need to decide for each field whether they are required or optional, and what restrictions should apply to them.

As of now, here are the features to be requested in the form:
- [x] first name & last name, both _required_
- [x] company, should be _optional_
- [x] website, should be _optional_
- [ ] picture, should be _optional_. There is no intent to put a storage for this, so this field should be an URL pointing towards a picture (_Not fully implemented yet_).
- [ ] checkbox to agree terms and conditions of Sedimark platform.

Current state of the form:
<img width="668" alt="image" src="https://github.com/user-attachments/assets/fce7533f-7b5e-422f-afce-730feded6ed1">

**Comment by xamcost** [comment]
@psotres @telsaleh Hi guys! I would actually suggest adding in this form a contact email. That doesn't cost us anything, and that would enable prospective consumers to contact providers in case they have questions on the offering. What do you think ?

**Comment by psotres** [comment]
Sorry @xamcost, I think I missed this issue.

To be honest right now I don't really know where that data would be stored for the consumers to retrieve it. It seems to be related with https://github.com/Sedimark/mvm-participant-onboarding/issues/4. Maybe @mikeus9908 and @alby0x0c can provide some extra input about the enrollment flow.

**Comment by psotres** [comment]
Hi all,

We have been having some discussions between UC and LINKS about this. Long post here 😆 

First, @mikeus9908 is still working on solving https://github.com/Sedimark/mvm-participant-onboarding/issues/3, which is needed to be able to fully define the enrolment flow and continue progressing with the participant onboarding sub-stream.

In parallel, we have been thinking about the information asked from participants and how consumers will actually retrieve it. TLDR: Participant identifiers are going to be their DID identifier URI (e.g. `did:iota:lnk:0x2b7ac707e71eda93135e1a457718f2d87f387ff66d62c8084ef06301110086ef#W1lJ9vr23XhK8kV8YTu27njdBBwjtHG0AWoNveHGD18`), and that is the only piece of provider-related information that is going to be included within the offering. It seems @telsaleh has defined the conccepts of `publisher` and `creator` in an offering. So far, we are 1005 sure DID's URIs will be the ones used for publishers (providers) because they will be the ones interacting in the SEDIMARK marketplace. Not sure about creators because they could not even be SEDIMARK participants.

Now the question becomes: how do consumers discover who is the provider? (as they can only start from the DID id, which is "a long hex string"). Again, from that string a DID document can be resolved (DLT booth exposes a method for that so the frontend does not need to cope with that). Let's take the DID example in https://github.com/Sedimark/dlt-booth/issues/7, which is not the one used in SEDIMARK but it serves the purpose for this topic:

```json
{
  "id": "did:web:did-server:company1",
  "@context": [
    "https://www.w3.org/ns/did/v1",
    {
      "@base": "did:web:did-server:company1"
    }
  ],
  "service": [
    {
      "id": "#identity-hub-url",
      "type": "IdentityHub",
      "serviceEndpoint": "http://company1:7171/api/identity/identity-hub"
    },
    {
      "id": "#dsp-url",
      "type": "DSPMessaging",
      "serviceEndpoint": "http://company1:8282/api/dsp"
    },
    {
      "id": "#self-description-url",
      "type": "SelfDescription",
      "serviceEndpoint": "http://company1:7171/api/identity/identity-hub/self-description"
    }
  ],
  "verificationMethod": [
    {
      "id": "#company1-key-1",
      "controller": "",
      "type": "EcdsaSecp256k1VerificationKey2019",
      "publicKeyJwk": {
        "kty": "EC",
        "kid": "yy_d7lbBzYC8EC0NDdX-r4fyf9SLsl8ZzviM5EzWRS4",
        "crv": "P-256",
        "x": "WohdFPABfA3tv5jYqR_qPEpZDRjSjQaoh4PlHBLemlY",
        "y": "c6eeeLzV0m4HEioTUL59yREOqILPxpx5m-MNCTDO6nI"
      }
    }
  ],
  "authentication": [
    "#company1-key-1"
  ]
}
```
The example DID (which comes from dataspaces domain) is a web-based one, which is stored on the participant premises, so GDPR is not a problem. However, DID documents used in SEDIMARK are stored in the Registry (actually, in the Tangle). Thus, we don't want to open the discussion about GDPR in a shared DLT infrastructure. For that reason no private info should be stored in the DID (which is not the objective of the DID, but could be a possibility). 

What we can do is to follow a similar approach and include a serviceEndpoint for retrieving the self-description, which will be provided by the participant itself. Whether a URL pointing to the participant domain should be considered to be private info or not is something I prefer not to think about 🤔.

Next question is how the self-description is formatted/presented. From the screenshot in https://github.com/Sedimark/marketplace-frontend/issues/4#issue-2611107239, there are information there which is currently not included on the SEDIMARK VC. What do we do with that info? Remember, this serves the purpose of building a human readable participant profile. For M2M communication the DID URI is enough. For this reason we believe this should be deferred to 2025. Anyway, we have identified 2 options:

1. One approach would be to include an extra service endpoint in the DID document pointing to that info (self-description) together with its signature with the identity key (maybe a hash is enough as the only one able to modify the DID is its owner). That way the self-description could be validated by consumers to be sure it has not been tampered or modified without the participant consent. The info included on the self-description can be whatever we think is interesting, as it will be stored within the toolbox on participant premises and exposed in a public endpoint in a similar way as the self-listing.
2. The other one is to include that info on the SEDIMARK VC and enable a mechanism (also using service endpoints on the DID doc) for the consumer to be able to ask for a "public VP" of the provider and use selective disclosure from the SEDIMARK VC to generate that "public VP".

I personally like the second option a bit more as it uses more standard mechanisms, but it is more complicated to implement. Its counterpart is the fact that "getting a VP from the provider" to build its public human-readable profile is "strange", as VPs are supposed to be presented and not retrieved. Main difference, and the benefit of the second approach is the fact that the SEDIMARK issuer "has validated" such info (only theoretically, no real validation of the claims is performed by the issuer).

**Comment by xamcost** [comment]
Hi there! Following yesterday's meeting, I am putting here the properties we decided to ask in the participant registration form. They will be exposed publicly via a json document hosted on the participant's premises. These properties are:

Property | Required | Comments
---  | --- | ---
First name | ❌ | 
Last name | ❌ |
Username | ✅ | May be stored in the catalogue to ease search by providers
Email | ❌ | The main way to contact a provider so far, maybe we will add more later
Organization | ❌ | 
Picture | ❌ | A URL to a picture/logo
Homepage | ❌ | A URL to an organization or personal website

Optionally, we could consider a _summary/short description_ and _description_ fields, to leave the option to participants to write a small bio about them. 

Feel free to suggest any addition/removal. 

@telsaleh @juanrasantana Let us know if we can help translating these fields into their equivalent in known ontologies like foaf.

**Comment by telsaleh** [comment]
Hi @psotres , @xamcost , @mikeus9908 ,

![Image](https://github.com/user-attachments/assets/33168172-c8bb-4408-96ba-25058409da36)

For the properties highlighted by @xamcost, we'll adopt most of them from schema.org except for "homepage" which will be from foaf.

**Comment by psotres** [comment]
Thanks @telsaleh !!

After the phone call we just had, let's stick to use DID as unique account identifier (e.g: https://www.w3.org/TR/did-1.0/#a-simple-example), which is guaranteed to be unique. In additon, let's use `alternateName` from `schema.org` as the required property during registration, taking into account that alternateName is not guaranteed to be unique, just a friendly representation of the DID.

**Comment by mikeus9908** [comment]
Hi @telsaleh!
I'm not familiar with ontology, so I'm sorry if my question is silly. May I ask you to provide an example on how to add the `schema.org` context for the properties that you've provided?

**Comment by telsaleh** [comment]
Hi @mikeus9908,

Sure, please see the example below. I have also added a link to the json-ld playground so you can see the different serializations, to see which fits your representation best.  --> [JSON-LD Playground](https://json-ld.org/playground/#startTab=tab-flattened&json-ld=%7B%22%40context%22%3A%7B%22sedi%22%3A%22https%3A%2F%2Fw3id.org%2Fsedimark%2Fontology%23%22%2C%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22foaf%22%3A%22http%3A%2F%2Fxmlns.com%2Ffoaf%2F0.1%2F%22%2C%22surrey%22%3A%22https%3A%2F%2Fsedimark.surrey.ac.uk%2F%22%7D%2C%22%40graph%22%3A%5B%7B%22%40id%22%3A%22surrey%3A0194f4b4-a039-7b1c-9dea-6b30b38239b8%22%2C%22%40type%22%3A%22schema%3APerson%22%2C%22schema%3AgivenName%22%3A%22Tarek%22%2C%22schema%3AfamilyName%22%3A%22Elsaleh%22%2C%22schema%3AaccountId%22%3A%22telsaleh%22%2C%22schema%3Aemail%22%3A%7B%22%40id%22%3A%22mailto%3Atarek%40surrey.ac.uk%22%7D%2C%22schema%3AmemberOf%22%3A%7B%22%40id%22%3A%22https%3A%2F%2Fcvssp.surrey.ac.uk%22%7D%2C%22schema%3Aimage%22%3A%7B%22%40id%22%3A%22https%3A%2F%2Fwww.surrey.ac.uk%2Fsites%2Fdefault%2Ffiles%2Fstyles%2Fmedium%2Fpublic%2F2020-09%2Fcvssp-logo-banner_0.png%3Fitok%3DNWySl2AY%22%7D%2C%22foaf%3Ahomepage%22%3A%7B%22%40id%22%3A%22https%3A%2F%2Fwww.surrey.ac.uk%2Fpeople%2Ftarek-elsaleh%22%7D%2C%22schema%3AalternateName%22%3A%7B%22%40id%22%3A%22Forza%20Juventus%22%7D%7D%5D%7D&frame=%7B%7D&context=%7B%22%40context%22%3A%7B%22sedi%22%3A%22https%3A%2F%2Fw3id.org%2Fsedimark%2Fontology%23%22%2C%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22foaf%22%3A%22http%3A%2F%2Fxmlns.com%2Ffoaf%2F0.1%2F%22%2C%22surrey%22%3A%22https%3A%2F%2Fsedimark.surrey.ac.uk%2F%22%7D%7D)

```json

{
  "@context": {
    "sedi": "https://w3id.org/sedimark/ontology#",
    "schema": "http://schema.org/",
    "foaf": "http://xmlns.com/foaf/0.1/",
    "surrey": "https://sedimark.surrey.ac.uk/"
  },
  "@graph": [
    {
      "@id": "surrey:0194f4b4-a039-7b1c-9dea-6b30b38239b8",
      "@type": "schema:Person",
      "schema:givenName": "Tarek",
      "schema:familyName": "Elsaleh",
      "schema:accountId": "telsaleh",
      "schema:email": {
        "@id": "mailto:tarek@surrey.ac.uk"
      },
      "schema:memberOf": {
        "@id": "https://cvssp.surrey.ac.uk"
      },
      "schema:image": {
        "@id": "https://www.surrey.ac.uk/sites/default/files/styles/medium/public/2020-09/cvssp-logo-banner_0.png?itok=NWySl2AY"
      },
      "foaf:homepage": {
        "@id": "https://www.surrey.ac.uk/people/tarek-elsaleh"
      },
      "schema:alternateName": {
        "@id": "Forza Juventus"
      }
    }
  ]
}
```

**Comment by xamcost** [comment]
Hi there! With the help of @mikeus9908 , we've deployed an issuer and DLT booth instance in our kubernetes cluster dedicated to Sedimark. Manifests are in the [issuer repo](https://github.com/Sedimark/sedimark-issuer) and [DLT booth repo](https://github.com/Sedimark/dlt-booth/tree/main).

One small confirmation we need to move on the frontend integration, especially for @psotres and @telsaleh . Since we just put `alternateName` in the DID upon onboarding, it means that after the creation of the participant identity, we also need to push its more complete self-description (including other optional fields like `name`, `homepage`, `email`, ...) somewhere... is this somewhere the offering-manager (which then pushes it to the catalogue), or directly in the catalogue ?

Thanks in advance!
